src/Avalonia.Themes.Fluent/Controls/ManagedFileChooser.xaml

```diff
@@ -0,0 +1,324 @@
+                        </ListBox.ItemTemplate>
+                    </ListBox>
+                    <DockPanel x:Name="NavBar" DockPanel.Dock="Top" Margin="8,5,8,0" VerticalAlignment="Center">
+                        <Rectangle Fill="{DynamicResource SystemControlHighlightAltBaseMediumLowBrush}" Height="1" Margin="0,5,0,0" DockPanel.Dock="Bottom"/>
+                        <DockPanel Margin="4,0">
+                            <Button Command="{Binding GoUp}" DockPanel.Dock="Left">
+                                <Path Data="M 0 7 L 7 0 L 14 7 M 7 0 L 7 16" Stroke="{Binding $parent[Button].Foreground}" StrokeThickness="1" HorizontalAlignment="Center" VerticalAlignment="Center" Margin="0,1,0,-1"/>
+                            </Button>
+                            <TextBox x:Name="Location" Text="{Binding Location}">
+                                <TextBox.KeyBindings>
+                                    <KeyBinding Command="{Binding EnterPressed}" Gesture="Enter"/>
+                                </TextBox.KeyBindings>
+                            </TextBox>
+                        </DockPanel>
+                    </DockPanel>
+                    <DockPanel Margin="8,0,8,5" DockPanel.Dock="Bottom">
+                        <Rectangle Fill="{DynamicResource SystemControlHighlightAltBaseMediumLowBrush}" Height="1" Margin="0,0,0,5" DockPanel.Dock="Top"/>
+                        <DockPanel Margin="4,0">
+                            <DockPanel DockPanel.Dock="Top" Margin="0,0,0,4">
+                                <ComboBox DockPanel.Dock="Right"
+                                    IsVisible="{Binding ShowFilters}" 
+                                    Items="{Binding Filters}"
+                                    SelectedItem="{Binding SelectedFilter}" />
+                                <TextBox Text="{Binding FileName}" Watermark="File name" IsVisible="{Binding !SelectingFolder}" />
+                            </DockPanel>
+                            <CheckBox IsChecked="{Binding ShowHiddenFiles}" Content="Show hidden files" DockPanel.Dock="Left"/>
+                            <UniformGrid x:Name="Finalize" HorizontalAlignment="Right" Rows="1">
+                                <Button Command="{Binding Ok}">OK</Button>
+                                <Button Command="{Binding Cancel}">Cancel</Button>
+                            </UniformGrid>
+                        </DockPanel>
+                    </DockPanel>
+
+                    <DockPanel Grid.IsSharedSizeScope="True">
+                        <Grid DockPanel.Dock="Top" Margin="15 5 0 0" HorizontalAlignment="Stretch">
+                            <Grid.ColumnDefinitions>
+                                <ColumnDefinition Width="20" SharedSizeGroup="Icon" />
+                                <ColumnDefinition Width="16" SharedSizeGroup="Splitter"  />
+                                <ColumnDefinition Width="400" SharedSizeGroup="Name" />
```

Comment:
Initial window size feels too big for me, as well as Name column is way bigger than average file name. Probably Window size can be changed to be smaller, and column width updated to fit everything.

Reply:
Name column default width is now 275, which brings the entirety of the other columns to just within the default window bounds.